scripts/run_design.py

```diff
@@ -23,7 +23,7 @@ def run_design(design_path, flow, error_flow, flow_args):
     design = Design(design_path)
 
     # Create temp folder
-    build_dir = Path.cwd() / "build" / flow / (design.path.relative_to(paths.DESIGNS_PATH))
+    build_dir = Path.cwd() / "build" / flow / design.path.name
```

Comment:
I think this will work to support designs outside of the examples directory, however it removes the current directory organization within the build directory.

Perhaps a better solution would be to check if it with the designs directory (https://stackoverflow.com/a/34236245/609215) and in that case leave the existing directory structure.  

If it outside of the designs directory, perhaps we could do something like:
`"build" / flow / "<external>" / design.path.name`

@dallinjdahl Thoughts?

Reply:
I think if we're going to move to support external designs with arbitrary file paths we should move the build artifacts next to each design. If we use the external tag, we risk overwriting designs with the same name but different paths.

Reply:
I agree <external> is a bit of a hack, but I also rather not put the generated files beside the source code. They did this in another project I'm a part of and I'm not a fan. It got really messy. 

In this case, I rather reconstruct the full absolute path within the build folder.

Reply:
given an arbitrary path `/path/to/source` we could derive `/path/to/build/artifact` such that all the artifacts are still in a separate build directory, but relative to the source rather than our tool.  That matches the way almost all tools work.  I don't know that I can think of a tool that depends on paths relative to the install directory...

Reply:
The build directory is not necessarily placed in the install directory.  It is placed in cwd (https://github.com/byuccl/bfasst/blob/d3bc3b1f3969326d38c72297072a3880ba3f9795/scripts/run_design.py#L26 -- although this script probably ought to have a command line option to override the location).